### PR TITLE
Add example revolving around Twitch data to documentation

### PR DESCRIPTION
**Pull request by kklein**
You can find a rendered version of the new example in the docs here:
https://datajudge--48.org.readthedocs.build/en/48/examples/example_twitch.html

### CODE REVIEW

File: docs/source/examples/example_twitch.rst [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+made available. We would like to use it but can't be sure of its validity just yet. As a consequence
+we would like to assess the quality of the data in version 2.
+
+The original data set can be found on `kaggle <https://www.kaggle.com/datasets/aayushmishra1512/twitchdata>`_.
+For the sake of this tutorial, we slightly process it and provide two version of it.
+One can either recreate this by executing this `processing <https://github.com/Quantco/datajudge/tree/main/docs/source/examples/twitch_upload.py>`_
+oneself on the original data or download our processed files (`version 1 <https://github.com/Quantco/datajudge/tree/main/docs/source/examples/twitch_version1.csv>`_
+and `version 2 <https://github.com/Quantco/datajudge/tree/main/docs/source/examples/twitch_version2.csv>`_) right away.
+
+The aforementioned script also populates two tables in a Postgres database. This resembles the following:
```
Comment (ivergara) [inline docs/source/examples/example_twitch.rst:19]:
```suggestion
The script above also populates two tables in a Postgres database. This resembles the following:
```

File: docs/source/examples/example_twitch.rst [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+     - English
+
+Note that we expect both version 1 and version 2 to follow this structure. Due to them
+being assembled at different points in time, merely their rows shows differ.
+
+
+Now let's write an actual specification, expressing our expectations against the data.
+First, we need to make sure a connection to database can be established at test execution
+time. How this is done exactly depends on how you set up your database. When using our
+default setup with running `$ ./start_postgres.sh <https://github.com/Quantco/datajudge/blob/main/start_postgres.sh>`_,
```
Comment (ivergara) [inline docs/source/examples/example_twitch.rst:100]: Probably here would be good to mention it's a container-based setup.

File: docs/source/examples/example_twitch.rst [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+    import sqlalchemy as sa
+
+
+    @pytest.fixture(scope="module")
+    def datajudge_engine():
+        address = os.environ.get("DB_ADDR", "localhost")
+        connection_string = f"postgresql://datajudge:datajudge@{address}:5432/datajudge"
+        return sa.create_engine(connection_string)
+
+Once a way to connect to database is defined, we want to declare our data sources and
```
Comment (ivergara) [inline docs/source/examples/example_twitch.rst:116]:
```suggestion
Once a way to connect to the database is defined, we want to declare our data sources and
```

File: docs/source/examples/example_twitch.rst [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+express expectations against them. In this example we have two tables in the same database -
```
Comment (ivergara) [inline docs/source/examples/example_twitch.rst:117]:
```suggestion
express expectations against them. In this example, we have two tables in the same database -
```

File: docs/source/examples/example_twitch.rst [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+one table per version of the Twitch data.
+
+
+Yet, let's start with a very simple example only using version 2. We want to use our
```
Comment (ivergara) [inline docs/source/examples/example_twitch.rst:121]:
```suggestion
Yet, let's start with a straightforward example only using version 2. We want to use our
```

File: docs/source/examples/example_twitch.rst [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+domain knowledge that values of the ``language`` column should only contain letters
```
Comment (ivergara) [inline docs/source/examples/example_twitch.rst:122]:
```suggestion
domain knowledge that constrains the values of the ``language`` column only to contain letters
```

File: docs/source/examples/example_twitch.rst [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+
+   /usr/local/Caskroom/miniconda/base/envs/datajudge/lib/python3.10/site-packages/datajudge/pytest_integration.py:25: AssertionError
+   ===================================== short test summary info ======================================
+   FAILED twitch_specification.py::test_func[VarCharRegex::tempdb.public.twitch_v2] - AssertionError...
+   FAILED twitch_specification.py::test_func[KolmogorovSmirnov2Sample::public.twitch_v1 | public.twitch_v2]
+   FAILED twitch_specification.py::test_func[UniquesEquality::public.twitch_v1 | public.twitch_v2]
+   FAILED twitch_specification.py::test_func[NumericMean::public.twitch_v2 | public.twitch_v2] - Ass...
+   =================================== 4 failed, 4 passed in 1.80s ====================================
+
+So we see that we might not want to trust version 2 of the data as is. What exactly do we
```
Comment (ivergara) [inline docs/source/examples/example_twitch.rst:309]:
```suggestion
So we see that we might not want to blindly trust version 2 of the data as is, and that a new exploration might be in order. What exactly do we
```

File: docs/source/examples/twitch_upload.py [hunk 8]
```diff
@@ -0,0 +1,52 @@
+import os
+
+import numpy as np
+import pandas as pd
+import sqlalchemy as sa
+
+SEED = 1337
+
+# Data obtaines from
```
Comment (ivergara) [inline docs/source/examples/twitch_upload.py:9]:
```suggestion
# Data obtained from
```

**Review by ivergara — APPROVED** [review]
Excellent job!

File: docs/source/examples/example_twitch.rst (hunk 2 above)
Comment (kklein) [reply docs/source/examples/example_twitch.rst]: Good point! I restructured this a bit to talk about the dockerized setup at the very beginning.

File: docs/source/examples/example_twitch.rst (hunk 7 above)
Comment (kklein) [reply docs/source/examples/example_twitch.rst]: I was unfamiliar with this usage of the expression 'in order' and took the liberty to rephrase slightly.

**Comment by kklein** [comment]
Thanks a lot for the blazingly fast review @ivergara :)

**Pull request merged**